websockets-jsr/src/main/java/io/undertow/websockets/jsr/WebSocketSessionRemoteEndpoint.java

```diff
@@ -273,50 +275,55 @@ public void sendText(final String partialMessage, final boolean isLast) throws I
             if(partialMessage == null) {
                 throw JsrWebSocketMessages.MESSAGES.messageInNull();
             }
-            if (binaryFrameSender != null) {
-                throw JsrWebSocketMessages.MESSAGES.cannotSendInMiddleOfFragmentedMessage();
-            }
-            if (textFrameSender == null) {
-                textFrameSender = undertowSession.getWebSocketChannel().send(WebSocketFrameType.TEXT);
-            }
-            try {
-                Channels.writeBlocking(textFrameSender, WebSocketUtils.fromUtf8String(partialMessage));
-                if(isLast) {
-                    textFrameSender.shutdownWrites();
+
+            synchronized (this) {
+                if (binaryFrameSender != null) {
+                    throw JsrWebSocketMessages.MESSAGES.cannotSendInMiddleOfFragmentedMessage();
+                }
+                if(textFrameSender == null) {
+                    textFrameSender = undertowSession.getWebSocketChannel().send(WebSocketFrameType.TEXT);
                 }
-                Channels.flushBlocking(textFrameSender);
-            } finally {
-                if (isLast) {
-                    textFrameSender = null;
+                try {
+                    Channels.writeBlocking(textFrameSender, WebSocketUtils.fromUtf8String(partialMessage));
```

Comment:
Whole section starting with Channels.writeBlocking() and ending with Channes.flushBlocking() must be executed outside of intrinsic lock section otherwise it will introduce deadlock possibility.

Reply:
@ropalka That's a good comment! 👍 
I will change the section so that there is no impact on the intrinsic lock.

Reply:
@ropalka 
As you mentioned, the section `"Channels.writeBlocking() ~ Channels.flushBlocking()"` was excluded from the intrinsic lock range.

Please review again.